freebsd-{rescue,usbin}-9.9999 needs /usr/include/arpa/telnet.h

diff --git a/include/arpa/Makefile b/include/arpa/Makefile
index 8a7f63d..a480b31 100644
--- a/include/arpa/Makefile
+++ b/include/arpa/Makefile
@@ -1,12 +1,7 @@
 # $FreeBSD$
 
-.include <bsd.own.mk>
-
 NO_OBJ=
-INCS=	ftp.h inet.h nameser.h nameser_compat.h tftp.h
-.if ${MK_TELNET} == "no"
-INCS+=	telnet.h
-.endif
+INCS=	ftp.h inet.h nameser.h nameser_compat.h telnet.h tftp.h
 INCSDIR=${INCLUDEDIR}/arpa
 
 .include <bsd.prog.mk>
