diff -Nur baselayout-2.1.orig/Makefile baselayout-2.1/Makefile
--- baselayout-2.1.orig/Makefile	2011-10-01 04:49:52.000000000 +0900
+++ baselayout-2.1/Makefile	2012-06-03 03:04:53.000000000 +0900
@@ -70,8 +70,10 @@
 	touch $(DESTDIR)/var/tmp/.keep
 	install -m 1777 -d $(DESTDIR)/tmp
 	touch $(DESTDIR)/tmp/.keep
+ifeq ($(OS),Linux)
 	ln -snf /run $(DESTDIR)/var/run
 	ln -snf /run/lock $(DESTDIR)/var/lock
+endif
 	# FHS compatibility symlinks stuff
 	ln -snf /var/tmp $(DESTDIR)/usr/tmp
 

