--- Makefile.orig	2009-08-02 19:56:31.000000000 +0200
+++ Makefile	2009-08-24 00:12:27.000000000 +0200
@@ -3,24 +3,24 @@
 
 .include <bsd.own.mk>
 
-SUBDIR=	ac accton acpitools altq apm apmd arp bad144 bind bootp \
+SUBDIR=	ac accton acpitools altq apm apmd arp bad144 bootp \
 	btattach btconfig btdevctl bthcid btpand catman \
-	chown chroot chrtbl cnwctl cpuctl cron dev_mkdb \
+	chown chroot chrtbl cnwctl cpuctl dev_mkdb \
 	dhcp diskpart dumpfs dumplfs edquota eeprom \
-	envstat eshconfig etcupdate extattrctl fssconfig fwctl gpioctl \
-	grfconfig grfinfo gspa hilinfo ifwatchd inetd installboot \
+	envstat eshconfig extattrctl fssconfig fwctl gpioctl \
+	grfconfig grfinfo gspa hilinfo ifwatchd installboot \
 	iopctl iostat ipwctl irdaattach isdn iteconfig iwictl\
-	kgmon lastlogin link lmcconfig lockstat lpr mailwrapper makefs \
+	kgmon lastlogin link lmcconfig lockstat mailwrapper makefs \
 	map-mbone mdconfig memswitch mlxctl mmcformat mopd mountd moused \
 	mrinfo mrouted mscdlabel mtrace \
-	mtree ndbootd ndiscvt netgroup_mkdb nfsd ntp ofctl paxctl pcictl \
+	mtree ndbootd ndiscvt netgroup_mkdb nfsd ofctl paxctl pcictl \
 	pppd psrset pstat pwd_mkdb postinstall powerd quot quotacheck \
 	quotaon rarpd rbootd rdate \
 	repquota rmt rpc.bootparamd rpc.lockd rpc.pcnfsd \
  	rpc.statd rpcbind rwhod sa screenblank sdpd services_mkdb sesd \
 	schedctl sliplogin sntp \
-	spray srtconfig sti sunlabel sup syslogd tadpolectl tcpdchk \
-	tcpdmatch tcpdump tcpdrop timed tpctl traceroute trpt unlink \
+	spray srtconfig sti sunlabel sup syslogd tadpolectl \
+	tcpdrop timed tpctl traceroute trpt unlink \
 	usbdevs user videomode vipw veriexecgen vnconfig wiconfig wlanctl wpa \
 	wsconscfg wsfontload wsmoused wsmuxctl zdump zic
 
